### PR TITLE
Update to latest conformance image

### PR DESCRIPTION
**Pull request by smaye81**
This updates the repo to use the latest conformance image, which uses the new organization on Dockerhub as well as the new proto packages (`connectrpc.conformance.v1` instead of `grpc.testing`).

As a part of this update, this attempts a fix at decoding the `ErrorDetails` value, which is returned as a base64-encoded string. The Connect protocol states that implementations should emit unpadded values and that downstream implementations should accept both padded and unpadded values. This adds padding to this string if needed as part of the `ConnectError` decode logic.

### CODE REVIEW

File: Libraries/Connect/Interfaces/ConnectError.swift [hunk 1]
```diff
@@ -73,6 +73,23 @@ public struct ConnectError: Swift.Error, Sendable {
...
+                withPad: "=",
+                startingAt: 0
+            )
+
+            self.init(
+                type: try container.decodeIfPresent(String.self, forKey: .type) ?? "",
+                payload: Data(base64Encoded: padded)
+            )
+        }
+
```
Comment (smaye81) [inline Libraries/Connect/Interfaces/ConnectError.swift:92]: @rebello95 Here is the padding fix I mentioned.

File: Libraries/Connect/Interfaces/ConnectError.swift [hunk 2]
```diff
@@ -73,6 +73,23 @@ public struct ConnectError: Swift.Error, Sendable {
             case payload = "value"
         }
 
+        public init(from decoder: Swift.Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+
+            // Read the base64-encoded payload and then pad if needed
+            let encodedPayload = try container.decode(String.self, forKey: .payload)
```
Comment (rebello95) [inline Libraries/Connect/Interfaces/ConnectError.swift:80]: Should this be a `decodeIfPresent`? This is changing behavior to throw if `payload` is omitted

File: Libraries/Connect/Interfaces/ConnectError.swift (hunk 1 above)
Comment (rebello95) [reply Libraries/Connect/Interfaces/ConnectError.swift:92]: Do tests cover receiving both a padded and unpadded error?

File: Libraries/Connect/Interfaces/ConnectError.swift [hunk 3]
```diff
@@ -73,6 +73,23 @@ public struct ConnectError: Swift.Error, Sendable {
...
+            let padded = encodedPayload.padding(
+                toLength: ((encodedPayload.count+3)/4)*4,
```
Comment (rebello95) [inline Libraries/Connect/Interfaces/ConnectError.swift:82]:
```suggestion
                toLength: ((encodedPayload.count + 3) / 4) * 4,
```
Also, this is a bit magic -- can you inline an explanation for this?

File: Libraries/Connect/Interfaces/ConnectError.swift (hunk 1 above)
Comment (smaye81) [reply Libraries/Connect/Interfaces/ConnectError.swift:92]: Do you mean the conformance suite? If so, no, they don't account for that. But probably a good one to add when it's refactored.

File: Libraries/Connect/Interfaces/ConnectError.swift (hunk 2 above)
Comment (smaye81) [reply Libraries/Connect/Interfaces/ConnectError.swift]: Definitely should. Fixed.

File: Libraries/Connect/Interfaces/ConnectError.swift [hunk 4]
```diff
@@ -73,6 +73,28 @@ public struct ConnectError: Swift.Error, Sendable {
...
+                // Then, pad the string to that length
+                toLength: ((encodedPayload.count + 3) / 4) * 4,
+                withPad: "=",
+                startingAt: 0
+            )
+
+            self.init(
+                type: try container.decodeIfPresent(String.self, forKey: .type) ?? "",
+                payload: Data(base64Encoded: padded)
+            )
```
Comment (rebello95) [inline Libraries/Connect/Interfaces/ConnectError.swift:95]:
```suggestion
            let paddedPayload = encodedPayload.padding(
                // Calculate the nearest multiple of 4 that is >= the length of encodedPayload,
                // then pad the string to that length.
                toLength: ((encodedPayload.count + 3) / 4) * 4,
                withPad: "=",
                startingAt: 0
            )
            self.init(
                type: try container.decodeIfPresent(String.self, forKey: .type) ?? "",
                payload: Data(base64Encoded: paddedPayload)
            )
```

**Review by rebello95 — APPROVED** [review]

**Review by rebello95 — APPROVED** [review]

**Comment by rebello95** [comment]
Would be nice to add tests for unpadded `ConnectError`s in a follow-up PR

**Comment by smaye81** [comment]
Agreed. I'll put a PR together for that. Thanks for reviewing!

**Pull request merged**